File: frontend/src/services/dashboardService.ts
import apiClient from './apiClient';
import { ApiResponse } from '@/types/api.types';
import { DashboardResponse, GunlukHarcamaResponse, KategoriHarcamaResponse } from '@/types/dashboard.types';

export const dashboardService = {
  async getDashboard(ay: number, yil: number): Promise<ApiResponse<DashboardResponse>> {
    const response = await apiClient.get<ApiResponse<DashboardResponse>>('/dashboard', { params: { ay, yil } });
    return response.data;
  },

  async getGunlukHarcamalar(baslangic: string, bitis: string): Promise<ApiResponse<GunlukHarcamaResponse[]>> {
    const response = await apiClient.get<ApiResponse<GunlukHarcamaResponse[]>>('/dashboard/gunluk', { params: { baslangic, bitis } });
    return response.data;
  },

  async getKategoriDagilimi(ay: number, yil: number): Promise<ApiResponse<KategoriHarcamaResponse[]>> {
    const response = await apiClient.get<ApiResponse<KategoriHarcamaResponse[]>>('/dashboard/kategoriler', { params: { ay, yil } });
    return response.data;
  },
};

File: backend/src/main/java/com/scinar/giderlerim/service/DashboardService.java
package com.scinar.giderlerim.service;

import com.scinar.giderlerim.dto.response.ApiResponse;
import com.scinar.giderlerim.dto.response.DashboardResponse;
import com.scinar.giderlerim.dto.response.GunlukHarcamaResponse;
import com.scinar.giderlerim.dto.response.KategoriHarcamaResponse;

import java.time.LocalDate;
import java.util.List;

public interface DashboardService {
    ApiResponse<DashboardResponse> getDashboard(Long kullaniciId, int ay, int yil);
    ApiResponse<List<GunlukHarcamaResponse>> getGunlukHarcamalar(Long kullaniciId, LocalDate baslangic, LocalDate bitis);
    ApiResponse<List<KategoriHarcamaResponse>> getKategoriDagilimi(Long kullaniciId, int ay, int yil);
}
